packages/webapp/pages/posts/[id].tsx

```diff
@@ -86,6 +87,12 @@ const PostPage = ({ id, initialData }: Props): ReactElement => {
     options: { initialData, retry: false },
   });
 
+  useEffect(() => {
+    if (!post?.private || !post?.permalink) return;
+
+    globalThis?.location.replace(post.permalink);
```

Comment:
I think the router is usually used for internal navigation, and keep the native when going outside the app. WDYT?

Reply:
@sshanzel Let's stick with this 🙏

Reply:
This redirect (permalink) is toward the actual link of the private link post, which is external, I believe.

`globalThis` is actually much safer to be used when loaded with NextJS as it avoids the app to crash like doing a regular `window` call. So I kinda prioritize using it over just a `window`.

Reply:
I am late to the party but `window` will always be defined inside even handlers and useEffect since it will be executed after hydration. Even with `globalThis` you don't need `?.` in those cases.

`globalThis` though is supposed to be a new standard for global object access and browser support is north of 90% so I agree with Lee that we should use instead of `window` especially since we have cross platform apps (extension and web).

Reply:
Ah yes totally agree with you guys, not in favor of window, the only alternative was using the router, but this makes sense for external links 🥳

Reply:
@capJavert where do you read all these things, I should subscribe to some RSS feed to it 😂